code/modules/mob/living/basic/farm_animals/rabbit.dm

```diff
@@ -0,0 +1,111 @@
+	attack_verb_simple = "kick"
+	butcher_results = list(/obj/item/food/meat/slab = 1)
+	ai_controller = /datum/ai_controller/basic_controller/rabbit
+	/// passed to animal_varity as the prefix icon.
+	var/icon_prefix = "rabbit"
+
+/mob/living/basic/rabbit/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/pet_bonus, "hops around happily!")
+	AddElement(/datum/element/animal_variety, icon_prefix, pick("brown", "black", "white"), TRUE)
+	if(prob(20)) // bunny
+		name = "bunny"
+
+/datum/ai_controller/basic_controller/rabbit
+	ai_traits = STOP_MOVING_WHEN_PULLED
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(/datum/ai_planning_subtree/random_speech/rabbit)
+
+/// The easter subtype of rabbits, will lay eggs and say Eastery catchphrases.
+/mob/living/basic/rabbit/easter
+	icon_state = "easter_rabbit_white"
+	icon_living = "easter_rabbit_white"
+	icon_dead = "easter_rabbit_white_dead"
+	icon_prefix = "easter_rabbit"
+	ai_controller = /datum/ai_controller/basic_controller/rabbit/easter
+	///passed to the egg_layer component as how many eggs it starts out as able to lay.
+	var/initial_egg_amount = 10
+	///passed to the egg_layer component as how many eggs it's allowed to hold at most.
+	var/max_eggs_held = 8
+
+/mob/living/basic/rabbit/easter/Initialize(mapload)
+	. = ..()
+	//passed to the egg_layer component as how many eggs it gets when it eats something.
+	var/eggs_added_from_eating = rand(1, 4)
+	var/list/feed_messages = list("[p_they()] nibbles happily.", "[p_they()] noms happily.")
+	AddComponent(/datum/component/egg_layer,\
+		/obj/item/surprise_egg,\
+		list(/obj/item/food/grown/carrot),\
+		feed_messages,\
+		list("hides an egg.","scampers around suspiciously.","begins making a huge racket.","begins shuffling."),\
+		initial_egg_amount,\
+		eggs_added_from_eating,\
+		max_eggs_held,\
+	)
+
+/datum/ai_controller/basic_controller/rabbit/easter
+	planning_subtrees = list(/datum/ai_planning_subtree/random_speech/rabbit/easter)
+
+/// Same deal as the standard easter subtype, but these ones are able to brave the cold of space with their handy gas mask.
+/mob/living/basic/rabbit/easter/space
+	icon_state = "space_rabbit_white"
+	icon_living = "space_rabbit_white"
+	icon_dead = "space_rabbit_white_dead"
+	icon_prefix = "space_rabbit"
+	ai_controller = /datum/ai_controller/basic_controller/rabbit/easter/space
+	// Minimum Allowable Body Temp, zero because we are meant to survive in space and we have a fucking RABBIT SPACE MASK.
+	var/minimum_survivable_temperature = 0
+	// Maximum Allowable Body Temp, 1500 because we might overheat and die in said RABBIT SPACE MASK.
+	var/maximum_survivable_temperature = 1500
+
+/mob/living/basic/rabbit/easter/space/Initialize(mapload)
+	. = ..()
+	// Cached List of suitable atmospherics conditions that we can survive in to pass to atmos_requirements element.
+	var/list/habitable_atmos = string_assoc_list(list(
```

Comment:
can't this be static?

Reply:
i copied this behavior from melbert in an earlier PR so i could figure out the implmentation and it reads
```dm
	// String assoc list returns a cached list, so this is like a static list to pass into the element below.
```

I'll rework the documentation.